email_notification/helpers.py

```diff
@@ -0,0 +1,38 @@
+from typing import List
+
+from permissions.helpers import has_unit_permission
+from reservations.models import Reservation
+from spaces.models import Unit
+from users.models import ReservationNotification, User
+
+
+def get_notification_recipients(
+    reservation: Reservation, notification_settings: List[ReservationNotification]
+) -> List[str]:
+    notification_recipients: List[User] = []
+    reservation_units = reservation.reservation_unit.all()
+    units = Unit.objects.filter(reservationunit__in=reservation_units)
+
+    # Get all users and collect the ones that can manage relevant units,
+    # have matching notification setting are not the reservation creator
+    users = User.objects.all()
```

Comment:
we could restrict the amount of user objects by filtering them by having some roles. Unit roles, service sector roles etc. and then we can go through "relevant" user instances.

Reply:
I was thinking about optimizing this, but because I'm such a mold head and didn't add a TODO comment, I forgot to do  it 🤦 

But now it is there! I didn't try to optimize the DB query to the max, but it should return much less users. Final checks are done in the code just like before.